src/VecSim/vec_sim.cpp

```diff
@@ -11,7 +11,7 @@
 #include "memory.h"
 
 extern "C" void VecSim_SetTimeoutCallbackFunction(timeoutCallbackFunction callback) {
-    VecSimIndexAbstract::setTimeoutCallbackFunction(callback);
+    VecSimIndexAbstract<float>::setTimeoutCallbackFunction(callback);
 }
```

Comment:
This function is called once from RediSearch. should it handle all template types? perhaps we should move the static callback to the interface and avoid the template here?